core/Concurrency/DistributedList.php

```diff
@@ -62,8 +62,12 @@ public function getAll()
      */
     public function setAll($items)
     {
-        foreach ($items as &$item) {
-            $item = (string)$item;
+        foreach ($items as $key => &$item) {
+            if (is_array($item)) {
+                throw new \InvalidArgumentException("Array item encountered in DistributedList::setAll() [ key = $key ].");
```

Comment:
maybe it's worth dumping the array in the logger as well ?
(note: i've already merged PR)